src/rdb/rdb_raft.c

```diff
@@ -3253,3 +3277,14 @@ rdb_raft_free_request(struct rdb *db, crt_rpc_t *rpc)
 		D_ASSERTF(0, DF_DB": unexpected opc: %u\n", DP_DB(db), opc);
 	}
 }
+
+void
+rdb_raft_module_init(void)
+{
+	raft_set_log_level(RAFT_LOG_DEBUG);
```

Comment:
I can't remember if we discussed this before, and whether there is planned a subsequent daos change for it: how will a daos admin choose to have only RAFT_LOG_INFO level logging emitted?

Reply:
@kccain, by setting log_mask to DEBUG for rdb and DD_MASK to include md.

I plan to change `daos` so that when `dmg server set-logmasks` is invoked, we will call `raft_set_log_level` accordingly (e.g., if md is off, then set `raft` log level to ERROR, if io is on, set `raft` log level to DEBUG). This will be mostly a (pre-mature?) performance optimization. Currently, with or without this PR #13770, `raft` *always* calls the `log` callback, although `daos` drops some of such messages based on DD_MASK. With the planned change, `raft` will not construct or call `log` for messages that `daos` will drop. Does not look like a high-priority item at the moment though.

Reply:
OK yes I see, thank you for that explanation. A complication with the potential future performance optimization I suppose is that if raft_set_log_level() is set to something less verbose (say RAFT_LOG_INFO) and later on dmg server set-logmasks is run with -d to change the DD_MASK streams (say, to enable io) then it will not have an effect on raft level logging that will continue to invoke the callback only for those raft log messages at RAFT_LOG_INFO level but not RAFT_LOG_DEBUG.

Reply:
@kccain, I see the situation like this:

- At the moment, we call `raft_set_log_level` once and request the most verbose log level. As a result, `raft` generates all log messages (same as without this PR), while `daos` drops **some of** them according to `daos`'s log level (better than without this PR).
- In the future, we call `raft_set_log_level` not only here, but also **during** `dmg server set-logmasks`. For instance. I've tried this when writing this PR, but I couldn't find the right support from the `daos` logging code.